.circleci/config.yml

```diff
@@ -0,0 +1,44 @@
+# Use the latest 2.1 version of CircleCI pipeline process engine.
+# See: https://circleci.com/docs/configuration-reference
+version: 2.1
+
+# Define a job to be invoked later in a workflow.
+# See: https://circleci.com/docs/configuration-reference/#jobs
+jobs:
+  lint:
+    docker:
+      - image: cimg/python:3.10.4
+    # Add steps to the job
+    # See: https://circleci.com/docs/configuration-reference/#steps
+    steps:
+      - checkout
+      - run:
+          name: "install requirements"
+          command: "pip install -r requirements.txt"
+      - run:
+          name: "run lint"
+          command: "black ./ --check"
+  test:
+    # Specify the execution environment. You can specify an image from Docker Hub or use one of our convenience images from CircleCI's Developer Hub.
+    # See: https://circleci.com/docs/configuration-reference/#executor-job
+    docker:
+      - image: cimg/python:3.10.4
+    # Add steps to the job
+    # See: https://circleci.com/docs/configuration-reference/#steps
+    steps:
+      - checkout
+      - run:
+          name: "install requirements"
+          command: "pip install -r requirements.txt"
+      - run:
+          name: "run tests"
+          command: "pytest"
```

Comment:
```suggestion
          command: |
            . venv/bin/activate
            pytest
```

Reply:
i tried the changes but in visual studio it had a bunch of red